changelog.d/4081.bugfix

```diff
@@ -0,0 +1 @@
+Fix race condition in populating reserved users
```

Comment:
could you try and expand on this a bit, please? what are reserved users, and where are they being populated?